taipy/core/_scheduler/_scheduler.py

```diff
@@ -33,17 +34,62 @@ class _Scheduler(_AbstractScheduler):
 
     """
 
-    def __init__(self, job_config: JobConfig = None):
-        super().__init__()
+    jobs_to_run: Queue = Queue()
+    blocked_jobs: List = list()
+    _dispatcher = _JobDispatcher(Config.job_config.nb_of_workers)  # type: ignore
+    lock = Lock()
+
+    @classmethod
+    def _set_nb_of_workers(cls, job_config: JobConfig = None):
         if not job_config:
             job_config = Config.job_config
-        self.jobs_to_run: Queue[Job] = Queue()
-        self.blocked_jobs: List[Job] = []
-        self._dispatcher = _JobDispatcher(job_config.nb_of_workers)  # type: ignore
-        self.lock = Lock()
 
+        cls._dispatcher._set_executer_and_nb_available_workers(job_config.nb_of_workers)  # type: ignore
+
+    @classmethod
+    def _recover_jobs(cls):
+        res = list()
+        blocked_or_submitted_jobs = list()
+        jobs = _JobManager._get_all()
+        jobs.sort(key=lambda x: x.creation_date)
+
+        for job in jobs:
+            if job.status == Status.RUNNING or job.status == Status.PENDING:
+                cls.__set_pending_job(job)
```

Comment:
why _set_pending_job_ with the _RUNNING_ status?

Reply:
if it was running before (during the last taipy run) and was stopped, we will put it to pending to prepare to be run again. Does it make sense to you?

Reply:
Totally thanks!